src/rulesets/asyncapi/index.json

```diff
@@ -0,0 +1,398 @@
+      "description": "AsyncAPI object should contain `license` object.",
+      "recommended": true,
+      "type": "style",
+      "given": "$",
+      "then": {
+        "field": "info.license",
+        "function": "truthy"
+      }
+    },
+    "asyncapi2-operation-description": {
+      "description": "Operation `description` must be present and non-empty string.",
+      "recommended": true,
+      "type": "style",
+      "given": [
+        "$.channels.*.[publish,subscribe]"
+      ],
+      "then": {
+        "field": "description",
+        "function": "truthy"
+      }
+    },
+    "asyncapi2-operation-operationId": {
+      "description": "Operation should have an `operationId`.",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.channels.*.[publish,subscribe]"
+      ],
+      "then": {
+        "field": "operationId",
+        "function": "truthy"
+      }
+    },
+    "asyncapi2-parameter-description": {
+      "description": "Parameter objects should have a `description`.",
+      "recommended": false,
+      "type": "style",
+      "given": [
+        "$.components.parameters.*",
+        "$.channels.*.parameters.*"
+      ],
+      "then": {
+        "field": "description",
+        "function": "truthy"
+      }
+    },
+    "asyncapi2-payload-default": {
+      "description": "Default must be valid against its defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits[?(@.schemaFormat === void 0)].payload.default^",
+        "$.components.messages[?(@.schemaFormat === void 0)].payload.default^",
+        "$.channels.*.[publish,subscribe][?(@property === 'message' && @.schemaFormat === void 0)].payload.default^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "default",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-payload-examples": {
+      "description": "Examples must be valid against their defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits[?(@.schemaFormat === void 0)].payload.examples^",
+        "$.components.messages[?(@.schemaFormat === void 0)].payload.examples^",
+        "$.channels.*.[publish,subscribe][?(@property === 'message' && @.schemaFormat === void 0)].payload.examples^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "$.examples.*",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-payload-unsupported-schemaFormat": {
+      "description": "Message schema validation is only supported with default unspecified `schemaFormat`.",
+      "severity": 2,
```

Comment:
let's use "warn" "error" "info" for things so folks (incouding me) dont need to work it out? :D

Reply:
@philsturgeon Fixed.

I've tried to do the same for oas (#1099), but it'll require some more work.
I'll table #1099 until this one is merged.